src/main/minio.js

```diff
@@ -2036,6 +2038,63 @@ export class Client {
         .on('end', () => cb(null, bucketNotification))
     })
   }
+
+  // Listens for bucket notifications. Returns an EventEmitter.
+  listenBucketNotification(bucketName, notificationARN) {
+    if (!isValidBucketName(bucketName)) {
+      throw new errors.InvalidBucketNameError(`Invalid bucket name: ${bucketName}`)
+    }
+    if (typeof notificationARN !== 'string') {
+      throw new TypeError('notificationARN must be of type string')
+    }
+
+    let emitter = new EventEmitter()
+
+    // Listening constitutes repeatedly requesting s3 whether or not any
+    // changes have occurred. If we receive an 'end' event, stop.
+
+    let end = false
+    emitter.on('end', () => { end = true })
```

Comment:
> I don't think emitter.removeListener() is idiomatic to stop notifications — that's just to remove the listener for notifications.

that is right.

'end' event to stop the notification does not sound right, because here library is like a producer of events and application using it a consumer of events though event emitter does not put restrictions on who can emit events. A better API to stop notifications is emitter.stop() along the lines of [FSWatcher.close()](https://nodejs.org/api/fs.html#fs_watcher_close)



> And since we start the polling loop with process.nextTick, all listeners should be attached by the time we start the GET request.

Though we can't assume that the application will immediately attach a listener usually that's how it is so we need not give a `start()`

but the `process.nextTick` is not adding value here because the GET response will always be on a different tick, so it can be removed.